packages/core/src/tenants/index.ts

```diff
@@ -16,17 +17,26 @@ export class TenantPool {
   });
 
   async get(tenantId: string): Promise<Tenant> {
-    const tenant = this.cache.get(tenantId);
-
-    if (tenant) {
-      return tenant;
+    const tenantPromise = this.cache.get(tenantId);
+
+    if (tenantPromise) {
+      const { createdAt, wellKnownCache } = await tenantPromise;
+      const expiresAt = await wellKnownCache.get(
+        'tenant-cache-expires-at',
+        WellKnownCache.defaultKey
+      );
+
+      if (!expiresAt || expiresAt < createdAt) {
+        return tenantPromise;
+      }
+      // Otherwise, create a new tenant
```

Comment:
to be more specific, "create a new tenant instance"

Reply:
updated